### DISCUSSION TITLE
[Feature] WebDav shares

### DISCUSSION THREAD

**Discussion by jon6fingrs** [immich-app/immich#7488] [Feature Request]
### The feature

A lot of electronic picture frames, photo screensavers, etc are able to access a WebDAV folder. 
I think it would be a helpful addition if the smart albums based on people or places could be shared as a virtual WebDAV share. That way you could leverage the awesome AI and sorting abilities of immich and benefit from a fairly ubiquitous file protocol.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Duplicate of #1687

**Comment by Mannshoch** [comment]
This could be nice to have e.g. a virtual Directory that reflect e.g. a Tag or an Album. It would allow to setup a collection in immich, But Linux and Windows could present the images to normal application which could work with that file.

- Apps could read the files
- Apps could upload files, which itself get tagged at least with the same atributes that created the webdav share. (Optional)
- Apps could update or remove a file. (Optional)
In that cases I would add the possibility to have to approve the changes in immich and do not update directly.
  - "Inside this webdav share this images got deleted, peas approve or abort". (Optional)
  - "Inside this webdav share this images got updated, please compare and approve or discard". (Optional)

**Comment by jon6fingrs** [comment]
I know this isn't exactly what having virtual webdav shares would do, but I made a script to download a number of random images from an album or person id with certain filter options. This doesn't help with uploading images back to the server, but at least provides some number of images that can be put through a slideshow or something and then rerunning the script on a schedule keeps rotating the images so that there are always new ones.

https://github.com/jon6fingrs/immich-dl

Not sure the exact reason you wanted to be able to access files through a virtual webdav but this solves the problem for me.

**Reply by Mannshoch** [reply]
My support of this idea came from this case here:
https://github.com/immich-app/immich/discussions/15409

**Reply by jon6fingrs** [reply]
Got it. Yeah, not sure what the best solution would be. Seems like immich can do mostly what you want anyway using user interface and shared albums?
